frontend/src/components/NoteView/NoteContainer.tsx

```diff
@@ -6,19 +6,25 @@ import {
   NoteWrapper,
 } from "../styles/Note.styled";
 import NoteEditor from "./NoteEditor";
+import useDebounceSave from "../../hooks/useDebounceSave";
+import { Note } from "../../common/internal";
 
 interface NoteProps {
   setNoteScrolled: (arg: boolean) => void;
   setShouldShowTitleOnToolBar: (arg: boolean) => void;
   shouldShowEditorMenuBar: boolean;
+  selectedNote: Note;
 }
 
-function Note({
+function NoteContainer({
   setNoteScrolled,
   setShouldShowTitleOnToolBar,
   shouldShowEditorMenuBar,
+  selectedNote,
 }: NoteProps) {
-  const [title, setTitle] = useState("Untitled");
+  const [activeNote, setActiveNote] = useState(selectedNote);
```

Comment:
Avoid having a `Note` object as state. This will create unnecessary rerender in other components if you update one of the property.

It would be best to have separate states for each note fields: `title`, `htmlContent` and `plainTextContent` (or just use `editor` for content-related fields?)